docs/user-guide/monitoring.md

```diff
@@ -0,0 +1,172 @@
+# Prometheus and Grafana installation
+
+This tutorial will show you how to install [Prometheus](https://prometheus.io/) and [Grafana](https://grafana.com/) for scraping the metrics of the NGINX Ingress controller.
+
+!!! Important: this example uses `emptyDir` volumes for Prometheus and Grafana. This means once the pod dies you will lose all the data.
+
+## Before You Begin
+
+The NGINX Ingress controller should already be deployed according to the deployment instructions [here](../deploy/index.md).
+
+Note that the yaml files that are used in this tutorial, are stored in the [deploy/monitoring](https://github.com/kubernetes/ingress-nginx/tree/master/deploy/monitoring) folder of the GitHub repository [kubernetes/ingress-nginx](https://github.com/kubernetes/ingress-nginx).
+
+## Deploy and configure Prometheus Server
+
+The Prometheus server is needed to configure so that it can discover endpoints of services. If a Prometheus server is already running in the cluster and if it is configured in a way that it can find the ingress controller pods, no extra configuration is needed.
+
+If there is no existing Prometheus server running, the rest of this tutorial will guide you through the step needed to deploy a properly configured Prometheus server.
+
+The configuration file is provided by a `ConfigMap`.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-configuration
+  labels:
+    name: prometheus-configuration
+  namespace: ingress-nginx
+data:
+  prometheus.yml: |-
+    global:
+      scrape_interval: 10s
+    scrape_configs:
+    - job_name: 'ingress-nginx-endpoints'
+      kubernetes_sd_configs:
+      - role: pod
+        namespaces:
+          names:
+          - ingress-nginx
+
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
+        action: keep
+        regex: true
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
+        action: replace
+        target_label: __scheme__
+        regex: (https?)
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
+        action: replace
+        target_label: __metrics_path__
+        regex: (.+)
+      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+        action: replace
+        target_label: __address__
+        regex: ([^:]+)(?::\d+)?;(\d+)
+        replacement: $1:$2
+
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: prometheus-service
+        action: drop
+```
+
+```console
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/monitoring/configuration.yaml
+configmap "prometheus-configuration" created
+```
+
+The next yaml snippet is used to deploy Prometheus in Kubernetes:
+
+```yaml
```

Comment:
If this is a dupe of `deploy/monitoring/prometheus.yaml`, wouldn't it be better to just link it?

Reply:
done